cppwinrt/code_writers.h

```diff
@@ -1156,9 +1156,9 @@ namespace cppwinrt
     {%
         if constexpr (!std::is_same_v<D, %>)
         {
-            auto const& castedResult = static_cast<% const&>(static_cast<D const&>(*this));
+            auto const [castedResult, code] = static_cast<D const*>(this)->template try_as_with_reason<%>();
```

Comment:
I would argue its the & that was extraneous before (because it lifetime extended the temporary coming from a cast). Now it's easier to reason about (it's a normal return value that gets moved).

Reply:
This seems to have re-added a spurious copy (with AddRef/Release) so that is not good.  Re-adding the `&` seems to have removed the extra copy while still keeping the object alive until the end of the scope, when Release gets called at the expected time.  The spurious copy was adding significant binary size to one of the internal binaries I'm testing against.

Reply:
Now with `as` this `&` is probably not useful anymore (`as` returns by value)

Reply:
Seems like this didn't actually make a difference.  I'm still seeing the same binary size with and without the `&` here.

Reply:
Suggests that the & extends a temporary, i.e. it may not be useful.